pkg/network/sriov.go

```diff
@@ -0,0 +1,54 @@
+package network
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/render"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+)
+
```

Comment:
One piece is missing, we should check for SRIOV against OpenShift network operator. In case user enabled SRIOV in network-addons operator, we need to check that it is indeed enabled also in OpenShift.

We already have this logic in Multus module.

Reply:
@phoracek I think the assumption should be that their operator is NOT used for SR-IOV deployment, no? Because these two pieces are not compatible.

Reply:
What we do in Multus is: If openshift network operator is deployed and Multus is requested on addons operator, just make sure it is enabled on openshift operator (we don't deploy anything ourselves).

Since we cannot run both our and openshift SR-IOV plugins at the same time, I though we could follow the same logic. If we are deployed on OpenShift 4 and our user requests SR-IOV, make sure it is enabled on OpenShift (if not, then fail).

I know it is not perfect, do you have any better flow in mind?

Reply:
@phoracek do you mean 'Make sure it's NOT enabled on OpenShift'? Because if it's enabled there, we can't reasonably use what they deploy for SR-IOV support for KubeVirt, so arguably, in this case I would expect our operator to return an error and not proceed with deployment.

Reply:
Oh I see! Sounds good to me :)